woocommerce/woocommerce-ios
# [Product Creation AI] Update prompt drafting to reduce unexpected AI response

Closes: #11188 


## Description

Updates the product creation AI prompt to reduce the possibility of receiving unexpected extra response from AI. 

Changes
- Remove newline characters and empty spaces added for human readability and use a dictionary and JSON encoding to create the prompt. 
- Add a sentence to the prompt asking to send only JSON and nothing else.

In my testing, I see AI responding with unexpected responses sometimes. But we handle that by showing the error alert and asking the user to Retry. i.e. We don't treat that as a success case. Please report if you see any other issues. 

## Testing instructions
- Create a JN store with Jetpack AI plugin
- Login into the app
- Navigate to the Products tab
- Tap "+" -> "Create a product with AI"
- Follow the steps to enter the product name and features
- Validate that the AI generates the product details as expected
- Try the AI creation flow a few more times to generate diverse products and validate that you don't see an empty screen like in https://github.com/woocommerce/woocommerce-ios/issues/11188#issuecomment-1839949721 

## Screenshots

https://github.com/woocommerce/woocommerce-ios/assets/524475/86165972-40ec-4814-926e-01754b1c3479


---
